sdk/documenttranslator/ai-document-translator/src/documentTranslationClient.ts

```diff
@@ -0,0 +1,286 @@
+   * Note: If the server can't honor $top and/or $skip, the server must return an error to the client informing about it instead of just ignoring the query options.
+   * This reduces the risk of the client making assumptions about the data returned.
+   */
+  get(
+    options?: GetOperationDocumentsStatusParameters
+  ): Promise<
+    | GetOperationDocumentsStatus200Response
+    | GetOperationDocumentsStatus400Response
+    | GetOperationDocumentsStatus401Response
+    | GetOperationDocumentsStatus404Response
+    | GetOperationDocumentsStatus429Response
+    | GetOperationDocumentsStatus500Response
+    | GetOperationDocumentsStatus503Response
+  >;
+}
+
+export interface GetDocumentFormats {
+  /**
+   * The list of supported document formats supported by the Document Translation service.
+   * The list includes the common file extension, as well as the content-type if using the upload API.
+   */
+  get(
+    options?: GetDocumentFormatsParameters
+  ): Promise<
+    | GetDocumentFormats200Response
+    | GetDocumentFormats429Response
+    | GetDocumentFormats500Response
+    | GetDocumentFormats503Response
+  >;
+}
+
+export interface GetGlossaryFormats {
+  /**
+   * The list of supported glossary formats supported by the Document Translation service.
+   * The list includes the common file extension used.
+   */
+  get(
+    options?: GetGlossaryFormatsParameters
+  ): Promise<
+    | GetGlossaryFormats200Response
+    | GetGlossaryFormats429Response
+    | GetGlossaryFormats500Response
+    | GetGlossaryFormats503Response
+  >;
+}
+
+export interface GetDocumentStorageSource {
+  /** Returns a list of storage sources/options supported by the Document Translation service. */
+  get(
+    options?: GetDocumentStorageSourceParameters
+  ): Promise<
+    | GetDocumentStorageSource200Response
+    | GetDocumentStorageSource429Response
+    | GetDocumentStorageSource500Response
+    | GetDocumentStorageSource503Response
+  >;
+}
+
+export interface Routes {
+  (path: "/batches"): GetOperations;
+  (path: "/batches/{id}/documents/{documentId}", id: string, documentId: string): GetDocumentStatus;
+  (path: "/batches/{id}", id: string): CancelOperation;
+  (path: "/batches/{id}/documents", id: string): GetOperationDocumentsStatus;
+  (path: "/documents/formats"): GetDocumentFormats;
+  (path: "/glossaries/formats"): GetGlossaryFormats;
+  (path: "/storagesources"): GetDocumentStorageSource;
+}
+
+export type DocumentTranslationClient = Client & {
+  path: Routes;
+};
+
+export interface DocumentTranslationFactory {
+  (endpoint: string, credentials: TokenCredential | KeyCredential, options?: ClientOptions): void;
+}
+
+export default function DocumentTranslation(
+  endpoint: string,
+  credentials: TokenCredential | KeyCredential,
+  options: ClientOptions = {}
+): DocumentTranslationClient {
+  const baseUrl =
+    options.baseUrl ||
+    "{endpoint}/translator/text/batch/v1.0-preview.1".replace(/{endpoint}/g, endpoint);
```

Comment:
```suggestion
    `${endpoint}/translator/text/batch/v1.0-preview.1`
```

Reply:
This is a good idea, I'll try to change it this way

Reply:
Okay I remember why I used replace. It is because the generator gets ` "{endpoint}/translator/text/batch/v1.0-preview.1"` as a string we don't build it. So to get this we'd need to parse and modify the string to replace the parameter in it for the interpolation format

Reply:
hmm the only thing we have to change is replace `{endpoint}` with `${endpoint}` right? and making it start and end with ``` ` ```

Reply:
alright! I updated the generator do do this 😄